arangod/Agency/FailedLeader.cpp

```diff
@@ -233,7 +233,12 @@ bool FailedLeader::start(bool& aborts) {
   std::vector<std::string> planv;
   for (auto const& i : VPackArrayIterator(planned)) {
     auto s = i.copyString();
-    if (s != _from && s != _to) {
+    // _from and _to are added as first and last entries
+    // we can keep all others
+    // for security we will not use empty strings (empty servers should never happen)
+    // also we will not use any resigend servers in the list (this should never happen as well, but if it happens,
+    // this situation will repair itself by diverging replicationFactor.
+    if (s != _from && s != _to && !s.empty() && s[0] != '_') {
```

Comment:
Not using a resigned server here could lead to the fact that the former leader immediately drops all data although it could just restart fine.

Reply:
but as you wrote there, its not something that actually can happen...

Reply:
actually not:
The former leader is in `_to`, which will be added under all circumstances, so the data there is save.
This only guards against a different (earlier) resigned leader, that found it's way into the plan.
For this the state is actually unclear, because it will not find itself in the plan anymore, only the resigned variant.